Block/Checkout/ComponentSwitcherProcessor.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+namespace Bolt\Boltpay\Block\Checkout;
+
+use Bolt\Boltpay\Block\Checkout\LayoutProcessor;
+
+/**
+ * Class ComponentSwitcherProcessor
+ * Enable / disable components in the checkout page layout depending on the configuration.
+ *
+ * @package Bolt\Boltpay\Block\Checkout\Cart
+ */
+class ComponentSwitcherProcessor extends LayoutProcessor
+{
+    /**
+     * Process the layout
+     *
+     * @param array $jsLayout
+     * @return array
+     */
+    public function process($jsLayout) {
+        // Payment only checkout based on Magento default one-page checkpout
+        // (one-page checkout, Mageplaza one-step checkout etc.)
+        $jsLayout['components']['checkout']['children']['steps']['children']['billing-step']['children']['payment']
```

Comment:
is it possible any of it is undefined, like `$jsLayout['components']['checkout']['children']['steps']` is undefined?

Reply:
No, it is defined in our layout extension 
https://github.com/BoltApp/bolt-magento2/blob/one-step-checkout-support/view/frontend/layout/checkout_index_index.xml